aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -117,6 +123,135 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
+                │         └────────────────────────┘      └─────────────────┘     │
+                │                                                                 │
+                │                                                                 │
+                │                                                                 ▼
+        ┌─────────────────┐                                              ┌─────────────────────┐
+        │Update Secret API│────────────────────────────────────────────▶│ Return or Exception │
+        └─────────────────┘                                              └─────────────────────┘
+
+        Parameters
+        ----------
+        name: str
+            The ARN or name of the secret to add a new version to or create a new one.
+        value: str, dict or bytes
+            Specifies text data that you want to encrypt and store in this new version of the secret.
+        client_request_token: str, optional
+            This value helps ensure idempotency. Recommended that you generate
+            a UUID-type value to ensure uniqueness within the specified secret.
+            This value becomes the VersionId of the new version. This field is
+            auto-populated if not provided.
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the Secrets Manager update_secret API call
+
+        Raises
+        ------
+        SetSecretError
+            When attempting to update or create a secret fails.
+
+        Returns:
+        -------
+        SetSecretResponse:
+            The dict returned by boto3.
+
+        Example
+        -------
+        **Sets a secret***
+
+            >>> from aws_lambda_powertools.utilities import parameters
+            >>>
+            >>> parameters.set_secret(name="llamas-are-awesome", value="supers3cr3tllam@passw0rd")
+
+        **Sets a secret and includes an client_request_token**
+
+            >>> from aws_lambda_powertools.utilities import parameters
+            >>> import uuid
+            >>>
+            >>> parameters.set_secret(
+                    name="my-secret",
+                    value='{"password": "supers3cr3tllam@passw0rd"}',
+                    client_request_token=str(uuid.uuid4())
+                )
+
+        URLs:
+        -------
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/put_secret_value.html
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/create_secret.html
+        """
+
+        if isinstance(value, dict):
+            value = json.dumps(value)
+
+        if isinstance(value, bytes):
+            sdk_options["SecretBinary"] = value
+        else:
+            sdk_options["SecretString"] = value
+
+        if client_request_token:
+            sdk_options["ClientRequestToken"] = client_request_token
```

Comment:
I'm not seeing a default ClientRequestToken being generated as described on the docstring. Is it a bug? Or does the SDK generate one automatically?

Reply:
According to the AWS Documentation, it's autopopulated by boto3 if not provided.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/put_secret_value.html

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/create_secret.html